src/testing/vortex/java_driver/src/main/java/Main.java

```diff
@@ -0,0 +1,378 @@
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.HashMap;
+import java.util.Map;
+import com.tigerbeetle.AccountBatch;
+import com.tigerbeetle.Client;
+import com.tigerbeetle.IdBatch;
+import com.tigerbeetle.TransferBatch;
+import com.tigerbeetle.UInt128;
+
+/**
+ * A Vortex driver using the Java language client for TigerBeetle.
+ */
+public final class Main {
+  public static void main(String[] args) throws Exception {
+    assert args.length == 2 : "java driver requires two positional command-line arguments";
+
+    byte[] clusterID = UInt128.asBytes(Long.parseLong(args[0]));
+
+    var replicaAddressesArg = args[1];
+    String[] replicaAddresses = replicaAddressesArg.split(",");
+    if (replicaAddresses.length == 0) {
+      throw new IllegalArgumentException(
+          "REPLICAS must list at least one address (comma-separated)");
+    }
+
+    try (var client = new Client(clusterID, replicaAddresses)) {
+      var reader = new Driver.Reader(Channels.newChannel(System.in));
+      var writer = new Driver.Writer(Channels.newChannel(System.out));
+      var driver = new Driver(client, reader, writer);
+      while (true) {
+        driver.next();
+      }
+    }
+  }
+}
+
+record Driver(Client client, Reader reader, Writer writer) {
```

Comment:
Java so new that GitHub even fails to highlight it! Who needs flying cars when we have pattern matching in Java!